projects/extension/sql/idempotent/013-vectorizer-api.sql

```diff
@@ -596,3 +596,55 @@ select
   end as pending_items
 from ai.vectorizer v
 ;
+
+-------------------------------------------------------------------------------
+-- vectorizer_embed
+create or replace function ai.vectorizer_embed
+( vectorizer_id pg_catalog.int4
+, input_text pg_catalog.text
+, input_type pg_catalog.text default null
+) returns @extschema:vector@.vector
+as $func$
+declare
+    _config pg_catalog.jsonb;
+    _emb @extschema:vector@.vector;
+begin
+    select v.config operator(pg_catalog.->) 'embedding'
+    into strict _config
+    from ai.vectorizer v
+    where v.id operator(pg_catalog.=) vectorizer_id
+    ;
+
+    case _config operator(pg_catalog.->>) 'implementation'
+        when 'openai' then
+            _emb = ai.openai_embed
+            ( _config operator(pg_catalog.->>) 'model'
+            , input_text
+            , api_key_name=>(_config operator(pg_catalog.->>) 'api_key_name')
+            , dimensions=>(_config operator(pg_catalog.->>) 'dimensions')::pg_catalog.int4
+            , openai_user=>(_config operator(pg_catalog.->>) 'user')
+            );
+        when 'ollama' then
+            _emb = ai.ollama_embed
+            ( _config operator(pg_catalog.->>) 'model'
+            , input_text
+            , host=>(_config operator(pg_catalog.->>) 'base_url')
+            , keep_alive=>(_config operator(pg_catalog.->>) 'keep_alive')
+            , embedding_options=>(_config operator(pg_catalog.->) 'options')
+            );
+        when 'voyageai' then
+            _emb = ai.voyageai_embed
+            ( _config operator(pg_catalog.->>) 'model'
+            , input_text
+            , input_type=>coalesce(input_type, 'query')
+            , api_key_name=>(_config operator(pg_catalog.->>) 'api_key_name')
+            );
+        else
+            raise exception 'unsupported embedding implementation';
+    end case;
+
+    return _emb;
+end
+$func$ language plpgsql stable security invoker
```

Comment:
should we lie and make this immutable? My gut says yes. The embedding config shouldn't change for a vectorizer. Stable will vastly decrease the usefulness of this function

Reply:
actually instead of lying maybe make this take the jsonb of the config instead of the vector_id. Then it can really be immutable for real

Reply:
https://github.com/timescale/pgai/pull/289/commits/63269ee032fb93bc950e4bc59dc32dab6ebe3e44